flume-ng-node/src/main/java/org/apache/flume/node/Application.java

```diff
@@ -83,16 +84,25 @@ public synchronized void start() {
   }
 
   @Subscribe
-  public synchronized void handleConfigurationEvent(MaterializedConfiguration conf) {
-    stopAllComponents();
-    startAllComponents(conf);
+  public void handleConfigurationEvent(MaterializedConfiguration conf) {
+    if (stopping) {
+      logger.info("Will not handle Configuration Event while stopping");
+      return;
+    }
+    synchronized (this) {
+      stopAllComponents();
+      startAllComponents(conf);
+      
+    }
   }
 
   public synchronized void stop() {
+    stopping = true;
     supervisor.stop();
     if (monitorServer != null) {
       monitorServer.stop();
     }
+    stopping = false;
```

Comment:
In general it's a good practice to wrap this to a `finally` block. In this particular case it mightn't be necessary as this is only called from the shutdown hook, so in fact it doesn't matter whether this is set to `false`. But in that case do we really need this line, wdyt?